docs(block-node): add glossary and replace scattered Key Terms sections with inline links (#3142)
hiero-ledger/hiero-block-node: docs/block-node/glossary.md

See also: [Tier 1 Block Node](#tier-1-block-node), [Tier 2 Block Node](#tier-2-block-node).

### Block Node Type

***

The deployment shape of a Block Node, describing which services it provides and how much
history it retains. Types include [Rolling-History](#rolling-history),
[Full Node](#full-node), [Light Node](#light-node), [Private-Cloud](#private-cloud),
[Archive Server](#archive-server), and [Community Node](#community-node).
Not to be confused with [Block Node Tier](#tier-1-block-node).

### Block Number

***

A monotonically increasing integer assigned by consensus to each block produced by the
network. Block numbers start at 0 (genesis) and increment by 1 per block. Used in
`serverStatus` (`first_available_block`, `last_available_block`), subscription requests
(`start_block_number`, `end_block_number`), and backfill configuration.

### Block Proof

***

A cryptographic proof attached to each block that allows any consumer to independently
verify the block's authenticity using the network's [Ledger ID](#ledger-id) and
[WRAPS](#wraps) verification key. A block may contain multiple `BlockProof` items; they
always appear after the [Block Footer](#block-footer). Defined in

> [HIP-1056](https://hips.hedera.com/hip/hip-1056).

Reviewer: This definition only applies to the `TSSSignature` type of Block Proof.
RSA signature block proof types are verified with the RSA keys for the address book at the time the block was created. When we migrate the network to Post Quantum Cryptography (PQC) there will be new block proof types that are quantum-computation resistant.

A generic description might read as follows:

```suggestion
A cryptographic proof attached to each block that allows any consumer to independently
verify the block's authenticity. A block may contain multiple `BlockProof` items; they
always appear after the [Block Footer](#block-footer). Defined in
[HIP-1056](https://hips.hedera.com/hip/hip-1056).
```

HAPI version to detect when the [Cutover Boundary](#cutover-boundary) has been crossed
and automatically switch their data source from cloud storage to block-stream subscription.

### HIP (Hiero Improvement Proposal)

***

A formal specification document that proposes changes to the Hiero protocol, network
behaviour, or ecosystem standards. HIPs are the authoritative source for features such as
Block Streams ([HIP-1056](https://hips.hedera.com/hip/hip-1056)), Block Nodes
([HIP-1081](https://hips.hedera.com/hip/hip-1081)), on-chain registration
([HIP-1137](https://hips.hedera.com/hip/hip-1137)), TSS signatures
([HIP-1200](https://hips.hedera.com/hip/hip-1200)), and the record-to-block-stream cutover
([HIP-1193](https://hips.hedera.com/hip/hip-1193)). Browse all HIPs at
[hips.hedera.com](https://hips.hedera.com).

---

## J

### Jumpstart Data

***

A small binary file (`jumpstart.bin`, under 1.5 KB) containing the last wrapped block
number, its root hash, and the streaming Merkle tree state. Consensus Nodes load this file
at the release that activates WRB production, enabling them to resume block hash
calculation without scanning millions of historical blocks.

> See [WRB CLI Runbook](./operations/wrb-cli-runbook.md).

Reviewer: This changed late in the process. CN elected to use existing configuration mechanisms instead of a new file.

```suggestion
A small amount of configuration data (under 1.5 KB) containing the last wrapped block
number, its root hash, and the streaming Merkle tree state. Consensus Nodes are configured with this data
at the release that activates WRB production, enabling them to resume block hash
calculation without scanning millions of historical blocks.
See [WRB CLI Runbook](./operations/wrb-cli-runbook.md).
```

Consensus Nodes. Used in conjunction with [Solo](#solo-provisioner) to drive
realistic production-scale traffic against a Block Node before connecting it to the
live network. See [Load Testing with Solo and NLG](./operations/load-testing-a-deployed-block-node-using-solo-and-nlg.md).

---

## P

### Partial History

***

A service provided by some Block Nodes that make available a subset of the network's
history — for example, only the most recent 30 days, or blocks after a specific height.
See also: [Rolling-History](#rolling-history).

### Plugin

***

A composable unit of functionality in the Block Node. All major Block Node features
(verification, storage, publishing, subscribing, backfill, health checks) are implemented
as plugins conforming to the `BlockNodePlugin` interface. The set of active plugins
determines what services a deployed Block Node provides.
See [Architecture Overview](./architecture/architecture-overview.md).

### Private Archive

***

A service provided by some Block Nodes that store block stream data in an archive for the
benefit of a particular entity, rather than the network as a whole. Examples include cloud

> buckets, long-term tape, or replicated local disks.

Reviewer: This is slightly misleading, as the first "private archive" will be for the benefit of the Hedera public network.

```suggestion
A service provided by some Block Nodes that store block stream data in an archive with no
public access. This may be for the benefit of a private entity, or may be a form of disaster recovery support for the public network. Example storage locations include private cloud
buckets, long-term tape, or replicated local disks.
```

### Private-Cloud

***

A Block Node type deployed within a private network or cloud environment, typically
serving a single organisation's internal needs rather than the broader public network.
Defined in [Block Node Types and Tiers](./../../Block-Node-Types.md).

### Private Sphere

***

A Hiero network operated on behalf of a private entity.

### Publisher

***

An entity that publishes block data to a Block Node via the `publishBlockStream` gRPC API.
In a standard Hiero network, publishers are Consensus Nodes (for Tier 1 Block Nodes) or

> other Block Nodes forwarding the stream.

Reviewer: Block Nodes do not forward a stream to other block nodes via the Publish API.

```suggestion
In a typical Hiero network, publishers are Consensus Nodes.
```

state after downtime by requesting recent blocks and state snapshots from Block Nodes.
Service interfaces are defined; this capability is planned for a future release and is not
currently in active development.

### Rolling-History

***

A Block Node type that retains only recent history — for example, the last 24 hours or
30 days — rather than full history from genesis. Most Tier 2 nodes are expected to be
this type. Provides a [Partial History](#partial-history) service.
Defined in [Block Node Types and Tiers](./../../Block-Node-Types.md).

### Roster / TSS Roster

***

The current set of Consensus Nodes and their consensus weights used by the TSS scheme.
The genesis roster's hash is the [Ledger ID](#ledger-id). Use "Roster" in documentation
— "Address Book" is an older, largely obsolete term.

---

## S

### serverStatus

***

A gRPC endpoint (`BlockNodeService.serverStatus`) that returns metadata about a Block
Node's current state: `first_available_block`, `last_available_block`, and
`only_latest_state`. These three fields are the only fields on `ServerStatusResponse`.
The extended `ServerStatusDetailResponse` additionally carries `version_information` and

> `node_address_book`.

Reviewer: Detail response tends to expand, currently it's 6 fields.
Note also, the extended response does _not_ carry the fields returned in the base response.
```suggestion
The extended `ServerStatusDetailResponse` provides extended detail, including `version_information`,
`node_address_book`, `available_ranges`, `stored_ranges`, `tss_data` and `ranged_address_book_history`.
```

### `StateChanges`

***

A [Block Item](#block-item) type that carries explicit CRUD operations applied to the
network's named states (maps, queues, singletons) as part of a block. Block Nodes apply
`StateChanges` to maintain their local copy of network state. For batch transactions,
`StateChanges` appear at the batch boundary rather than per individual transaction.
Defined in [HIP-1056](https://hips.hedera.com/hip/hip-1056).

### Solo Provisioner

***

> The recommended tool for provisioning Block Nodes on cloud VMs or bare-metal Kubernetes
> clusters. Formerly known as Solo Weaver. Use "Solo Provisioner" in prose; the repository
> URL and filename retain `solo-weaver`.
> See [Deploy with Solo Provisioner](./operations/solo-weaver-single-node-k8s-deployment.md).

Reviewer: Part of this reads as instruction, and can probably be elided.
```suggestion
The recommended tool for provisioning Block Nodes on cloud VMs or bare-metal Kubernetes
clusters. Formerly known as Solo Weaver, and only referred to as `Solo Provisioner` in descriptions.
The repository URL and filenames may still contain references to `solo-weaver`.
See [Deploy with Solo Provisioner](./operations/solo-weaver-single-node-k8s-deployment.md).
```

### State Snapshot *(planned)*

***

A point-in-time capture of the complete network state (accounts, balances, smart contract
storage, and so on) at a specific block height. Block Nodes will produce state snapshots
from their locally managed state once state management is implemented. This capability is
planned for a future release.

### Subscriber

***

An entity that subscribes to a block stream from a Block Node via the
`subscribeBlockStream` gRPC API. Subscribers include Mirror Nodes, Tier 2 Block Nodes,
and any application consuming live or historical block data.
See also: [Publisher](#publisher).

---

## T

### Tier 1 Block Node

***

A Block Node that receives the block stream directly from one or more Consensus Nodes.
Tier 1 nodes are typically operated by trusted or Council-affiliated entities and form the
first layer of the block-stream distribution network.

### Tier 2 Block Node

***

A Block Node that receives the block stream from one or more upstream Block Nodes (Tier 1
or another Tier 2). Tier 2 nodes are permissionless and are commonly used to redistribute

> the stream to downstream clients such as Mirror Nodes.

Reviewer: Small expansion.

```suggestion
A Block Node that receives the block stream from one or more upstream Block Nodes (Tier 1
or another Tier 2). Tier 2 nodes are permissionless and are commonly used to redistribute
the stream to downstream clients such as Mirror Nodes or to provide value-added application-specific services.
```
